### DISCUSSION TITLE
Install Admin 4 gives composer problem

### DISCUSSION THREAD

**Discussion by gt1202** [vanilophp/framework#177] [General]
Followed the instructions for install: 

composer create-project --prefer-dist laravel/laravel vaniloapp ^11.0
cd vaniloapp
composer require vanilo/framework ^4.0
php artisan vendor:publish --provider="Konekt\Concord\ConcordServiceProvider" --tag=config

Then try to install admin and get an error:

composer req vanilo/admin ^4.0

Problem 1
    - Root composer.json requires vanilo/admin ^4.0, found vanilo/admin[dev-master, 4.0.x-dev (alias of dev-master)] but it does not match your minimum-stability.

How can I install admin?

**Comment by fulopattila122** [comment]
The **final** Vanilo 4 admin hasn't been released yet, but it's fairly stable, you can safely use it.

Add this line to your `composer.json` file to allow pre-release versions in your project:

```json
    "minimum-stability": "dev",
    "prefer-stable": true
```

> See also: https://stackoverflow.com/a/45019830/1016746 and https://getcomposer.org/doc/04-schema.md#minimum-stability

**Comment by gt1202** [comment]
Thanks for your reply.  I added the lines, but still get the same error.

`composer.json`:

```json
   "name": "laravel/laravel",
    "type": "project",
    "description": "The skeleton application for the Laravel framework.",
    "keywords": ["laravel", "framework"],
    "license": "MIT",
    "minimum-stability": "dev",
    "prefer-stable": true,
    "require": {
        "php": "^8.2",
        "laravel/framework": "^11.0",
        "laravel/tinker": "^2.9",
        "vanilo/framework": "^4.0"
    },
```

**Reply by fulopattila122** [reply]
What command do you run exactly?

**Comment by gt1202** [comment]
composer require vanilo/admin ^4.0

Running composer update vanilo/admin
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires vanilo/admin ^4.0, found vanilo/admin[dev-master, 4.0.x-dev (alias of dev-master)] but it does not match your minimum-stability.

**Reply by fulopattila122** [reply]
It's a known behavior of composer, which should've been solved by the "minimum-stability" setting.

Try the followings

1. Run `composer why-not vanilo/admin ^4.0` (it explains more details why it won't do it)
2. **Verify `composer.json` changes**: Ensure that the `"minimum-stability": "dev"` line was added correctly in the `composer.json` file. It should be at the root level of the JSON structure, not nested inside another key.
3. Review the composer.json File Structure: Make sure the structure of composer.json is valid JSON and that the minimum-stability **setting isn't overridden somewhere else in the file**.
4. **Clear Composer Cache**: Sometimes Composer might cache information about packages and ignore changes. You can advise your client to clear the Composer cache by running: `composer clear-cache`.
After clearing the cache, they should try running the require command again.
5. **Inspect the Composer Version**: The behavior of Composer can sometimes vary significantly between versions. Ensure your client is using a relatively current version of Composer. They can update Composer by running: `composer self-update`

If after trying these steps the issue persists, it could be useful to look at the exact composer.json file content to further diagnose the problem. Sometimes, the issue might be as simple as a typo or a JSON syntax error.

## As a LAST RESORT

> :warning: Running `composer update` is generally discouraged, so be sure to understand the caveats.

Check for `composer.lock` and the `vendor/` folder
- Delete the composer.lock* file and the vendor directory, then try running the install command again.
- Alternatively, running composer update might also help resolve this issue.

> \* Make sure to save a copy of the composer.lock file so that you can restore it if necessary

**Comment by gt1202** [comment]
I did a clean install, set "minimum-stability": "dev",  "prefer-stable": false and it installed successfully. Thanks for all your help.
